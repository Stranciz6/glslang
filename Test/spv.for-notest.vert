#version 450
layout(location=0) out highp int r;
void main() {
  int i;
  // This infinite loop results in bad SPIR-V generated, since the merge block
  // is dropped as unreachable.  It is still useful for testing the rest of the
  // code generation.
  for (i=0; ; i++) { r = i; }
}
