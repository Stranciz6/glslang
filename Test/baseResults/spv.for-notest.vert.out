spv.for-notest.vert
Warning, version 450 is not yet complete; most version-specific features are present, but some are missing.


Linked vertex stage:


// Module Version 10000
// Generated by (magic number): 80001
// Id's are bound by 22

                              Capability Shader
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              EntryPoint Vertex 4  "main" 14 20 21
                              Source GLSL 450
                              Name 4  "main"
                              Name 8  "i"
                              Name 14  "r"
                              Name 20  "gl_VertexID"
                              Name 21  "gl_InstanceID"
                              Decorate 14(r) Location 0
                              Decorate 20(gl_VertexID) BuiltIn VertexId
                              Decorate 21(gl_InstanceID) BuiltIn InstanceId
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeInt 32 1
               7:             TypePointer Function 6(int)
               9:      6(int) Constant 0
              13:             TypePointer Output 6(int)
           14(r):     13(ptr) Variable Output
              17:      6(int) Constant 1
              19:             TypePointer Input 6(int)
 20(gl_VertexID):     19(ptr) Variable Input
21(gl_InstanceID):     19(ptr) Variable Input
         4(main):           2 Function None 3
               5:             Label
            8(i):      7(ptr) Variable Function
                              Store 8(i) 9
                              Branch 10
              10:             Label
              15:      6(int) Load 8(i)
                              Store 14(r) 15
                              Branch 12
              12:             Label
              16:      6(int) Load 8(i)
              18:      6(int) IAdd 16 17
                              Store 8(i) 18
                              Branch 10
                              FunctionEnd
